src/storage/src/hummock/state_store.rs

```diff
@@ -599,8 +601,70 @@ impl StateStore for HummockStorage {
         }
     }
 
-    fn try_wait_epoch(&self, epoch: HummockReadEpoch) -> Self::WaitEpochFuture<'_> {
-        async move { Ok(self.local_version_manager.try_wait_epoch(epoch).await?) }
+    /// Waits until the local hummock version contains the epoch. If `wait_epoch` is `Current`,
+    /// we will only check whether it is le `sealed_epoch` and won't wait.
+    fn try_wait_epoch(&self, wait_epoch: HummockReadEpoch) -> Self::WaitEpochFuture<'_> {
+        async move {
+            // Ok(self.local_version_manager.try_wait_epoch(epoch).await?)
+            let wait_epoch = match wait_epoch {
+                HummockReadEpoch::Committed(epoch) => epoch,
+                HummockReadEpoch::Current(epoch) => {
+                    // let sealed_epoch = self.local_version.read().get_sealed_epoch();
+                    let sealed_epoch = (*self.seal_epoch).load(MemOrdering::SeqCst);
+                    assert!(
+                        epoch <= sealed_epoch
+                            && epoch != HummockEpoch::MAX
+                        ,
+                        "current epoch can't read, because the epoch in storage is not updated, epoch{}, sealed epoch{}"
+                        ,epoch
+                        ,sealed_epoch
+                    );
+                    return Ok(());
+                }
+                HummockReadEpoch::NoWait(_) => return Ok(()),
+            };
+            if wait_epoch == HummockEpoch::MAX {
+                panic!("epoch should not be u64::MAX");
+            }
+
+            let mut receiver = self.version_update_notifier_tx.subscribe();
+            let max_committed_epoch = *receiver.borrow();
```

Comment:
I think it is better to use `borrow_and_update` here to avoid unneccessary check in the loop if the value does not change.

Reply:
fix it